commands/remove.rb

```diff
@@ -13,10 +14,20 @@ def self.remove(pkg, verbose)
       return
     end
 
-    # Don't remove any of the packages ruby (and thus crew) needs to run.
-    if CREW_ESSENTIAL_PACKAGES.include?(pkg.name)
-      puts "Refusing to remove essential package #{pkg.name}.".lightred
-      return
+    # Determine dependencies of packages in CREW_ESSENTIAL_PACKAGES and
+    # their dependencies, as those are needed for ruby and crew to run,
+    # and thus should not be removed.
+    essential_deps = CREW_ESSENTIAL_PACKAGES.flat_map { |essential_pkg| Package.load_package("#{essential_pkg}.rb").get_deps_list }.push(*CREW_ESSENTIAL_PACKAGES).uniq.sort
+
+    if essential_deps.include?(pkg.name)
+      return if Package.load_package("#{pkg.name}.rb").in_upgrade
```

Comment:
we shouldn't need to load the package again, the loaded package object was already passed to `Command.remove`

```suggestion
      return if pkg.in_upgrade
```

Reply:
Thanks! Changed!